### PR TITLE
update spotify client

### PR DESCRIPTION
**Pull request by SarthakGupta22**
Get song features from the track id

### CODE REVIEW

File: src/api/spotify_client.py [hunk 1]
```diff
@@ -1,54 +1,96 @@
...
         return access_token
+
+    def get_song_features(self, track_id: str) -> dict[str, Any]:
+        """
+        Fetches the audio features of a given Spotify track using its track ID.
+        Returns a dictionary containing the track's audio features.
+        """
+        
+        # Fetch the access token (if it's not available, make the request to get it)
+        access_token: str | None = self.get_access_token()
```
Comment (ShrivatsAgrawal) [inline src/api/spotify_client.py]: Check is self.access_token already exists. If not exists then only create new one. Or if access token has expired. Will prevent many api calls.

File: src/api/spotify_client.py [hunk 2]
```diff
@@ -1,54 +1,96 @@
...
+        response: requests.Response = requests.get(url, headers=headers)
+
+        # Parse and return the JSON response containing the audio features
+        audio_features: dict[str, Any] = response.json()
+
+        return audio_features
+
+
+# Example Usage
+spotify_client = SpotifyClient()
```
Comment (AbhinavAtrishi) [inline src/api/spotify_client.py]: `if __name__ == "__main__"`

**Review by AbhinavAtrishi — CHANGES_REQUESTED** [review]

File: src/api/spotify_client.py (hunk 1 above)
Comment (ShrivatsAgrawal) [reply src/api/spotify_client.py]: Also assign to self.access_token for use by other functions

**Review by ShrivatsAgrawal — APPROVED** [review]

**Review by sarthakkg — APPROVED** [review]
Ship it

**Pull request merged**